tests/openai/test_openai_agent_autolog.py

```diff
@@ -0,0 +1,245 @@
+import copy
+from contextlib import contextmanager
+from typing import Optional
+from unittest import mock
+
+import pytest
+from openai.types.chat.chat_completion import ChatCompletion, Choice
+
+import mlflow
+from mlflow.entities import SpanType
+
+from tests.openai.test_openai_autolog import client  # noqa: F401
+from tests.tracing.helper import get_traces
+
+try:
+    from swarm import Agent, Swarm
+except ImportError:
+    pytest.skip("OpenAI Swarm not installed", allow_module_level=True)
+
+
+@contextmanager
+def mock_openai(oai_client, expected_responses):
+    original = oai_client.chat.completions.create
+    expected_responses = copy.deepcopy(expected_responses)
+    with mock.patch.object(oai_client.chat.completions, "create") as mock_create:
+
+        def _mocked_create(*args, **kwargs):
+            # We need to call the original SDK function because OpenAI autolog patches
+            # it to generate a span for completion.
+            original(*args, **kwargs)
+            return expected_responses.pop(0)
+
+        mock_create.side_effect = _mocked_create
+
+        yield
```

Comment:
```suggestion
    expected_responses = copy.deepcopy(expected_responses)

    def _mocked_create(*args, **kwargs):
        # We need to call the original SDK function because OpenAI autolog patches
        # it to generate a span for completion.
        original(*args, **kwargs)
        return expected_responses.pop(0)

    with mock.patch.object(oai_client.chat.completions, "create", side_effect=_mocked_create) as mock_create:
        yield
        
        mock_create.assert_called()
```